### PR TITLE
move sm_app header to gunrock.h

### PR DESCRIPTION
**Pull request by Laurawly**
@neoblizz Could you review?

### CODE REVIEW

File: gunrock/app/sm/sm_app.cu [hunk 1]
```diff
@@ -245,10 +242,36 @@ double sm(
...
+double sm(
+    const int            num_nodes,
+    const int            num_edges,
+    const int           *row_offsets,
+    const int           *col_indices,
+    const unsigned long *edge_values,
+    const int            num_runs,
+          int           *subgraphs)
+{
+    return sm(num_nodes, num_edges, row_offsets, col_indices,
```
Comment (jowens) [inline gunrock/app/sm/sm_app.cu]: why have `num_runs` as an argument if it's hardcoded to 1?

Comment (Laurawly) [reply gunrock/app/sm/sm_app.cu]: I think here is only a default value. @neoblizz correct me if I'm wrong. Other apps do so as well.

**Comment by Laurawly** [comment]
I don't see the error message of codecov ci check.

**Comment by neoblizz** [comment]
> I don't see the error message of codecov ci check.

You can ignore the codecov error, it is saying that you added some code that doesn't get coverage during testing. But that is understandable because we don't have a ctest for c-interface.

**Comment by Laurawly** [comment]
> > I don't see the error message of codecov ci check.
> 
> You can ignore the codecov error, it is saying that you added some code that doesn't get coverage during testing. But that is understandable because we don't have a ctest for c-interface.

I see. So are we good in merging it?

**Comment by neoblizz** [comment]
> > > I don't see the error message of codecov ci check.
> > 
> > 
> > You can ignore the codecov error, it is saying that you added some code that doesn't get coverage during testing. But that is understandable because we don't have a ctest for c-interface.
> 
> I see. So are we good in merging it?

Yes, but let's take care of the template/non-template issue as well before we merge this.

**Comment by Laurawly** [comment]
@neoblizz I think we should do a separate or for that some other apps needs to do the same thing in order to make this consistent

**Pull request merged**